### ISSUE TITLE
WPF WeakEventManager usage 

### ISSUE THREAD

**Issue by ElijahReva** [reactiveui/ReactiveUI#1617]
**Do you want to request a *feature* or report a *bug*?**
Bug. Regression on https://github.com/reactiveui/ReactiveUI/issues/661

**What is the current behavior?**
`PropertyChangedEventManager.DeliverEvent` used when setting reactive property for WPF application that using .

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

Reproduction repo. Just step into RaiseProperyChanged
https://github.com/ElijahReva/reactiveui.bug

**What is the expected behavior?**
According to https://github.com/reactiveui/ReactiveUI/issues/661 we should not use WeakEventManager on WPF platform

**What is the motivation / use case for changing the behavior?**
Core MVVM PropertyChanged event stop being fired after some time.

**Which versions of ReactiveUI, and which platform / OS are affected by this issue? Did this work in previous versions of ReativeUI? Please also test with the [latest stable](https://www.nuget.org/profiles/reactiveui) and [development snapshot](https://reactiveui.net/docs/getting-started/installation/nuget-packages/development-package-feed)**

ReactiveUI 8.0.0
ReactiveUI.WPF 8.0.0

Not sure when this got broken. This could be broken as part of migration to netstandart2.0. There is no NET_45 compilation flag for it.

**Other information (e.g. stack traces, related issues, suggestions how to fix)**

https://github.com/reactiveui/ReactiveUI/issues/661
https://github.com/reactiveui/ReactiveUI/pull/666

**Known workaround**
I've workarounded by inheriting from ReactiveObject and explicitly specifying an event handler for IReactiveObject.
```
public abstract class ReactiveObjectWPF : ReactiveObject, IReactiveObject
    {
        public new event PropertyChangingEventHandler PropertyChanging;

        void IReactiveObject.RaisePropertyChanging(PropertyChangingEventArgs args)
        {
            var handler = this.PropertyChanging;
            if(handler != null) {
                handler(this, args);
            }
        }

        public new event PropertyChangedEventHandler PropertyChanged;

        void IReactiveObject.RaisePropertyChanged(PropertyChangedEventArgs args)
        {
            var handler = this.PropertyChanged;
            if (handler != null) {
                handler(this, args);
            }
        }
    }
```

**Comment by ghuntley** [comment]
Howdy, @ElijahReva - if we broke you as part of the upgrade. Know that we are sorry. I won't be able to look at this immediately but [have asked in Slack](http://reactiveui.net/slack) if folks who have some time spare could look into this for you. Thank-you for attaching a repro. Compilation flags are defined here https://github.com/reactiveui/ReactiveUI/blob/develop/src/Directory.build.targets#L5

**Comment by ghuntley** [comment]
Based off eyeballing the source code - it looks like a regression.

Can you please clone down ReactiveUI, change https://github.com/reactiveui/ReactiveUI/blob/develop/src/ReactiveUI/ReactiveObject.cs#L31 to `NET_461` - verify that it fixes your problem then send a PR. I'll do a release in the morning just for this.

**Comment by ghuntley** [comment]
Change it in these two places

![image](https://user-images.githubusercontent.com/127353/39760196-0ed831b0-5318-11e8-9a39-efd7f0d46eec.png)

**Comment by ElijahReva** [comment]
Thanks. Will do!

**Comment by mikesena** [comment]
Hi Guys;

I'm less experienced with WPF and ReactiveUI, but think this bug may be linked to behaviour I'm experiencing at the moment.

I had been using the alpha 117 version for a while without issue.  When I upgraded to 8.0.0's final release, I started having bindings fail after some use / time.  For instance, I have a command updating a property that uses 'RaiseAndSetIfChanged', which updated a UI component.  That command would work at the start, then break after aggressive use of the application.

Is this bug likely to be what is causing my issue too?  Is there any way I could confirm it?

I've rolled back to the alpha version I was on before and haven't experienced the same issue.

If it's not likely to be linked, I'll raise a ticket to try and have it investigated.
